tutorial/examples/faqs/last_clicked_button.py

```diff
@@ -1,35 +1,44 @@
+import json
+
 import dash
 import dash_html_components as html
 from dash.dependencies import Input, Output
 
 app = dash.Dash(__name__)
 
 app.layout = html.Div([
-    html.Button('Button 1', id='btn-1', n_clicks_timestamp='0'),
-    html.Button('Button 2', id='btn-2', n_clicks_timestamp='0'),
-    html.Button('Button 3', id='btn-3', n_clicks_timestamp='0'),
+    html.Button('Button 1', id='btn-1'),
+    html.Button('Button 2', id='btn-2'),
+    html.Button('Button 3', id='btn-3'),
     html.Div(id='container')
 ])
 
 
 @app.callback(Output('container', 'children'),
-              [Input('btn-1', 'n_clicks_timestamp'),
-               Input('btn-2', 'n_clicks_timestamp'),
-               Input('btn-3', 'n_clicks_timestamp')])
+              [Input('btn-1', 'n_clicks'),
+               Input('btn-2', 'n_clicks'),
+               Input('btn-3', 'n_clicks')])
 def display(btn1, btn2, btn3):
-    if int(btn1) > int(btn2) and int(btn1) > int(btn3):
-        msg = 'Button 1 was most recently clicked'
-    elif int(btn2) > int(btn1) and int(btn2) > int(btn3):
-        msg = 'Button 2 was most recently clicked'
-    elif int(btn3) > int(btn1) and int(btn3) > int(btn2):
-        msg = 'Button 3 was most recently clicked'
-    else:
+    ctx = dash.callback_context
+
+    if not ctx.triggered:
         msg = 'None of the buttons have been clicked yet'
+    else:
+        button_num = ctx.triggered[0]['prop_id'].split('.')[0].split('-')[1]
```

Comment:
Just get the last character:

```suggestion
        button_num = ctx.triggered[0]['prop_id'][-1]
```

Reply:
That doesn't quite work, the number is in the middle. I suppose I could do
`button_num = ctx.triggered[0]['prop_id'].split('.')[0][-1]`
but that only saves a few characters, at the expense of generality in case someone extends this to 10+ buttons

Reply:
regex way might be cleaner. 'btn-(\d+)\.', but I feel you can instead of getting the button number, just render the first part of  `btn-3.n_clicks` would be enough for demonstration.

Reply:
> just render the first part of `btn-3.n_clicks` would be enough for demonstration.

I like that - the id `btn-3` is what's most useful for developers -> https://github.com/plotly/dash-docs/pull/414/commits/0c5d158632fc833b994f374f4b39328045e468f0

Reply:
(also made a pretty little table in that commit ^^)